azurelinuxagent/common/protocol/hostplugin.py

```diff
@@ -140,6 +141,21 @@ def get_api_versions(self):
 
         return return_val
 
+    def get_vm_settings_request(self):
+        if not self.ensure_initialized():
+            raise ProtocolError("HostGAPlugin: Host plugin channel is not available")
+
+        url = URI_FORMAT_VM_SETTINGS.format(self.endpoint, HOST_PLUGIN_PORT)
+
+        headers = {
+            _HEADER_VERSION: API_VERSION,
+           _HEADER_CONTAINER_ID: self.container_id,
+           _HEADER_HOST_CONFIG_NAME: self.role_config_name,
+           _HEADER_CORRELATION_ID: str(uuid.uuid4())  # TODO: persist correlation ID
```

Comment:
Should trace the correlationId to help with debugging.

Reply:
yup, that's the "TODO" part